### PR TITLE
Make the guaranteed price reactive and add guaranteed amount check

### PR DESCRIPTION
**Pull request by RastogiAbhijeet**

### CODE REVIEW

File: lib/services/protection.ts [hunk 1]
```diff
@@ -17,8 +17,7 @@ export class ProtectionsService {
       .from(VIEWS.DISTINCT_EXPIRATION_DATES)
       .select("expiration_date")
       .match({ base: currency })
-      .order("expiration_date")
-      .gte("expiration_date", new Date().toUTCString());
```
Comment (DenisGorbachev) [inline lib/services/protection.ts:21]: Why did you remove this condition?

Comment (RastogiAbhijeet) [reply lib/services/protection.ts:21]: The function name is just supposed to return only expirations dates and not just expiration date > any date. 

For now I have added a filter on the component level. 

May be I can create another function in the service named 
`getExpirationDatesByCurrencyGteNow`

Comment (DenisGorbachev) [reply lib/services/protection.ts:21]: Yeah, for now It's OK to have a filter at the component level - just wanted to make sure there's a filter *somewhere*
